Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -2628,3 +2628,47 @@ function Remove-CustomRole {
     }
 }
 
+<#
+    .Synopsis
+    This function allows the customer to restart HA on a cluster. This might be needed to clear HA alarms.
+    .EXAMPLE
+    Once the function is imported: Restart-ClusterHA -cluster_name "Cluster-1"
+#>
+function Restart-ClusterHA {
+    [AVSAttribute(30, UpdatesSDDC = $true)]
+    param(
+        [Parameter(Mandatory = $true,
+            HelpMessage = "What cluster to restart HA on.")]
+        [String]
+        $cluster_name
+    )
+
+    # Get the cluster object
+    try {
+        $cluster = Get-Cluster -Name $cluster_name -ErrorAction Stop
+    }
+    catch {
+        throw "Cluster $cluster_name not found."
+    }
+
+    $ha_status = ""
```

Comment:
Shouldn't we evaluate whether HA is enabled first to ascertain what action should be taken?  As in, if previous run command failed for whatever reason, couldn't it be reasoned that our default action should be to turn on HA if we see that it is disabled?

Also, if we enable this, we should have monitoring in place to catch any SDDC's that have HA disabled for an extended period of time.  If nothing, but to have a point of reference should customer suffer from an actual HA event, but were disabled when such event occurred.

Reply:
Do we have this as an internal geneva action?

Reply:
This commandlet could be exposed as a Geneva Action if needed.

Reply:
@et1975 I've been asking if Geneva and Run commands could share code but hadn't found a way. How can I do that?

Reply:
@gungazoo  Let's take that to Teams, this is a public repo.

Reply:
@Zsoldier -- Since it is the customer cycling it, I would say continue if HA is disabled. We should have a monitor if HA is disabled for a period of time.